client/lxd_images.go

```diff
@@ -930,3 +951,28 @@ func (r *ProtocolLXD) ExportImage(fingerprint string, image api.ImageExportPost)
 	return op, nil
 
 }
+
+// GetServerSupportedFilters returns two lists: one with filters supported by server and second one with not supported.
+func (r *ProtocolLXD) GetServerSupportedFilters(filters []string, i interface{}) ([]string, []string) {
```

Comment:
This feels a bit odd. I think it'd probably be simpler to turn this into an internal `validateFilters(filters []string) error` which then errors out if missing the extension or if the filter syntax is invalid.

Ultimately it's going to be up to the caller of those functions to pass in valid filters and to the server to return an error when something unsupported is passed.

Reply:
I've moved this method to `lxc/utils`.